tests/pytests/test_contains.py

```diff
@@ -93,39 +91,42 @@ def testSanity(env):
-        env.expect('ft.search', index_list[i], '*o55*', 'LIMIT', 0, 0).equal([444])
-        env.expect('ft.search', index_list[i], '*oo55*', 'LIMIT', 0, 0).equal([333])
-        env.expect('ft.search', index_list[i], '*oo555*', 'LIMIT', 0, 0).equal([33])
-
-        # 23xx & x23x & xx23 - 2323
-        env.expect('ft.search', index_list[i], '*23*', 'LIMIT', 0, 0).equal([1196])
-        # 234x & x234
-        start = time.time()
-        env.expect('ft.search', index_list[i], '*234*', 'LIMIT', 0, 0).equal([80])
-        # print(time.time() - start)
-        start = time.time()
-        env.expect('ft.search', index_list[i], '*o23*', 'LIMIT', 0, 0).equal([444])
-        # print(time.time() - start)
-        env.expect('ft.search', index_list[i], '*oo23*', 'LIMIT', 0, 0).equal([333])
-        env.expect('ft.search', index_list[i], '*oo234*', 'LIMIT', 0, 0).equal([33])
+        env.expect('ft.search', index_list[i], '*oo*', 'LIMIT', 0, 0).equal([4000])
+        # 55x & x55 - 555
+        env.expect('ft.search', index_list[i], '*55*', 'LIMIT', 0, 0).equal([76])
+        # 555
+        env.expect('ft.search', index_list[i], '*555*', 'LIMIT', 0, 0).equal([4])
+        env.expect('ft.search', index_list[i], '*o55*', 'LIMIT', 0, 0).equal([44])
+        env.expect('ft.search', index_list[i], '*oo55*', 'LIMIT', 0, 0).equal([33])
+        env.expect('ft.search', index_list[i], '*oo555*', 'LIMIT', 0, 0).equal([3])
+
+        # 23x & x23
+        env.expect('ft.search', index_list[i], '*23*', 'LIMIT', 0, 0).equal([80])
+        # 234
+        env.expect('ft.search', index_list[i], '*234*', 'LIMIT', 0, 0).equal([4])
+        env.expect('ft.search', index_list[i], '*o23*', 'LIMIT', 0, 0).equal([44])
+        env.expect('ft.search', index_list[i], '*oo23*', 'LIMIT', 0, 0).equal([33])
+        env.expect('ft.search', index_list[i], '*oo234*', 'LIMIT', 0, 0).equal([3])
 
         # suffix
         env.expect('ft.search', index_list[i], '*oo234', 'LIMIT', 0, 0).equal([3])
-        env.expect('ft.search', index_list[i], '*234', 'LIMIT', 0, 0).equal([40])
-        env.expect('ft.search', index_list[i], '*13', 'LIMIT', 0, 0).equal([400])
+        env.expect('ft.search', index_list[i], '*234', 'LIMIT', 0, 0).equal([4])
+        env.expect('ft.search', index_list[i], '*13', 'LIMIT', 0, 0).equal([40])
 
     # test timeout
+    for i in range(item_qty, item_qty * 5):
+        pl.execute_command('HSET', 'doc%d' % i, 't', 'foo%d' % i)
+        pl.execute_command('HSET', 'doc%d' % (i + item_qty), 't', 'fooo%d' % i)
+        pl.execute_command('HSET', 'doc%d' % (i + item_qty * 2), 't', 'foooo%d' % i)
+        pl.execute_command('HSET', 'doc%d' % (i + item_qty * 3), 't', 'foofo%d' % i)
+        pl.execute()
+
     env.expect('ft.config', 'set', 'TIMEOUT', 1).ok()
     env.expect('ft.config', 'set', 'ON_TIMEOUT', 'RETURN').ok()
     env.expect('ft.search', index_list[0], 'foo*', 'LIMIT', 0, 0).error() \
```

Comment:
Should we leave these timeout tests? Will they not be flaky?

Reply:
It wasn't until now. I thought it tests some internal timeout handling similar to what we do on VecSim. In this case, I think it better to keep it for the coverage. If that's not the case, we can remove it.